go/src/koding/klient/machine/machinegroup/idset/idset.go

```diff
@@ -0,0 +1,66 @@
+package idset
+
+import (
+	"koding/klient/machine"
+)
+
+// Union returns a set of all elements in A and B sets.
+func Union(a, b []machine.ID) []machine.ID {
+	aset := make(map[machine.ID]struct{})
+	for i := range a {
+		aset[a[i]] = struct{}{}
+	}
+
+	res := make([]machine.ID, len(a))
+	copy(res, a)
+
+	for i := range b {
+		if _, ok := aset[b[i]]; !ok {
+			res = append(res, b[i])
+		}
+	}
+
+	return res
+}
+
+// Intersection returns a set that contains all elements of A that also belong
+// to B, but no other elements.
+func Intersection(a, b []machine.ID) []machine.ID {
+	aset := make(map[machine.ID]bool)
+	for i := range a {
+		aset[a[i]] = true
```

Comment:
ah, you want me to assign `struct{}{}` and then check `_, ok := ...; ok`. will add this change in one of next PRs